pylanetary/spectral/core.py

```diff
@@ -0,0 +1,808 @@
+          The array of spectral data to be fit.
+    fit_type : string
+          The type of line shape used to fit the spectra. Options are Gaussian, Lorentzian, Voigt, PseudoV, Moffat, and All. It is not recommended that you use the option of all for anything more than a single pixel.
+    outfile : string
+          The name of the saved output pickle file. The 'pickle' file extension is automatically included.
+    initial_fit_guess : string, optional
+          The name of a saved LMFIT result to use as an intitial guess instead of the LMFIT guess function. Default is None.
+    RMS : float, optional
+          The numerical value of the data's RMS. Default is None; if not provided, an estimate is calculated using Astropy's siga_clipped_stats (https://docs.astropy.org/en/stable/api/astropy.stats.sigma_clipped_stats.html)
+    SN : float
+          The Signal to noise ratio minimum, which is used to mask out noisy data. Any pixel with a S/N ratio lower than this calue will not be fit. Default is value is 1. 
+    
+    '''
+    
+    if SN == None:
+      SN = 1
+    
+    picklefile = open(outfile+'.pickle', 'wb')
+    print('Numerical results will be saved in the '+outfile+'.pickle file')
+    results = {'x':[],'y':[],'center':[],'std':[],'chi':[],'redchi':[]}
+    
+    for xpix in range (0,self.xpixmax,1):
+      for ypix in range (0,self.ypixmax,1): 
+        s_n = abs(np.max(datas[xpix,ypix])/RMS)
+        if s_n > SN:
+          print('Modeling spectrum at pixel '+str(xpix)+','+str(ypix))
+          spectra = Spectrum(spec = datas[xpix,ypix], freq = self.xaxis, rest_f = 345.79598990, x_space = self.x_space, RMS = RMS) 
+          peak, std, chi, chi_red = spectra.fit_profile(fit_type = 'Moffat', initial_fit_guess = initial_fit_guess, showplots = False)
+          results['x'].append(xpix)
+          results['y'].append(ypix)
+          results['center'].append(peak)
+          results['std'].append(std) 
+          results['chi'].append(chi)
+          results['redchi'].append(chi_red)
+        elif s_n <= SN or np.isnan(s_n): 
+          results['x'].append(xpix)
+          results['y'].append(ypix)
+          results['center'].append(np.NAN)
+          results['std'].append(np.NAN) 
+          results['chi'].append(np.NAN)
+          results['redchi'].append(np.NAN)
+          
+    lenx = np.shape(datas)[0]
+    leny = np.shape(datas)[1]
+    
+    results['x']=np.reshape(results['x'],(lenx,leny))
+    results['y']=np.reshape(results['y'],(lenx,leny))
+    results['center']=np.reshape(results['center'],(lenx,leny))
+    results['std']=np.reshape(results['std'],(lenx,leny))
+    results['chi']=np.reshape(results['chi'],(lenx,leny))
+    results['redchi']=np.reshape(results['redchi'],(lenx,leny))
+      
+    pickle.dump(results,picklefile)
+    picklefile.close()
+    
+  ###############################################################################  
+  
+  def wind_calc(self, picklefile, restfreq, outfile):
```

Comment:
this function currently does not use self.anything; might be an indication it should exist outside of the Cube object?